letter-sender/src/test/java/com/christmas/letter/sender/service/SnsServiceTest.java

```diff
@@ -0,0 +1,170 @@
+package com.christmas.letter.sender.service;
+
+import com.christmas.letter.sender.exceptions.InvalidDataException;
+import com.christmas.letter.sender.exceptions.LetterValidator;
+import com.christmas.letter.sender.model.Letter;
+import com.christmas.letter.sender.model.Location;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.mockito.junit.jupiter.MockitoExtension;
+import software.amazon.awssdk.services.sns.SnsClient;
+import software.amazon.awssdk.services.sns.model.PublishRequest;
+import software.amazon.awssdk.services.sns.model.PublishResponse;
+
+import static org.hamcrest.MatcherAssert.assertThat;;
+
+import static org.hamcrest.Matchers.notNullValue;
+import static org.junit.Assert.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+class SnsServiceTest {
```

Comment:
The service has a new name, let's also rename this test.

Reply:
updated to LetterServiceTest